src/sources/host_metrics.rs

```diff
@@ -751,42 +747,57 @@ pub fn init_roots() {
 }
 
 impl FilterList {
-    fn is_empty(&self) -> bool {
-        self.includes.is_none() && self.excludes.is_none()
-    }
-
-    fn contains_str(&self, value: &str) -> bool {
-        (match &self.includes {
+    fn contains_str(&self, value: Option<&str>) -> bool {
+        (match (&self.includes, value) {
             // No includes list includes everything
-            None => true,
+            (None, _) => true,
+            // Includes list matched against empty value returns false
+            (Some(_), None) => false,
             // Otherwise find the given value
-            Some(includes) => includes.iter().any(|pattern| pattern.matches_str(value)),
-        }) && match &self.excludes {
+            (Some(includes), Some(value)) => {
+                includes.iter().any(|pattern| pattern.matches_str(value))
+            }
+        }) && match (&self.excludes, value) {
             // No excludes, list excludes nothing
-            None => true,
+            (None, _) => true,
+            // No value, never excluded
+            (_, None) => true,
```

Comment:
I think this branch (and the parallel line in `contains_path`) should read `(Some(_), None)`